crates/sargon/src/wrapped_radix_engine_toolkit/high_level/manifest_building/account_locker/manifest_account_locker.rs

```diff
@@ -0,0 +1,151 @@
+use crate::prelude::*;
+use radix_engine_interface::blueprints::locker::{
+    ACCOUNT_LOCKER_CLAIM_IDENT, ACCOUNT_LOCKER_CLAIM_NON_FUNGIBLES_IDENT,
+};
+
+impl From<AccountAddress> for ScryptoComponentAddress {
+    fn from(value: AccountAddress) -> ScryptoComponentAddress {
+        ScryptoComponentAddress::new_or_panic(value.node_id().0)
+    }
+}
+
+impl TransactionManifest {
+    pub fn account_locker_claim(
+        locker_address: &LockerAddress,
+        claimant: &AccountAddress,
+        claimable_resources: Vec<AccountLockerClaimableResource>,
+    ) -> Self {
+        let mut builder = ScryptoManifestBuilder::new();
+        let claimant_arg: ScryptoComponentAddress = (*claimant).into();
+        let claimable_resources =
+            Self::build_claimable_batch(claimable_resources, 50);
+
+        for claimable in claimable_resources.clone() {
```

Comment:
have you run `cargo clippy --fix --allow-staged` ? I think we should change `for claimable in claimable_resources.clone() ` to `for claimable in claimable_resources.iter()` no? (which will be references to `AccountLockerClaimableResource` so you might need to change the match a little bit, e.g. doing `.clone()` below).

Or if you are fine with consuming `claimable_resources` you can `into_iter()` and then all good, which requires you to do the `!claimable_resources.is_empty()` check before calling `into_iter()`. 

i.e.:
`
`
```rust
let should_deposit = !claimable_resources.is_empty();
for claimable_resource in claimable_resources.into_iter {
 /// code here should remain the same you have it
}
```

Reply:
I assume cloning an entire vector is less efficient than cloning each element from the same vector?